nvflare/private/fed/client/client_app_runner.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from nvflare.apis.fl_constant import FLContextKey
+from nvflare.apis.workspace import Workspace
+from nvflare.private.defs import EngineConstant
+from nvflare.private.fed.client.client_json_config import ClientJsonConfigurator
+from nvflare.private.fed.client.client_run_manager import ClientRunManager
+from nvflare.private.fed.client.client_runner import ClientRunner
+from nvflare.private.fed.client.client_status import ClientStatus
+from nvflare.private.fed.client.command_agent import CommandAgent
+
+
+class ClientAppRunner:
```

Comment:
What is the difference/purpose for each of these classes:

"ClientRunManager" / "ClientRunner" / "ClientAppRunner"

If they mean similar thing can we just merge them?

Reply:
"ClientRunManager" is the client engine (ClientEngineExecutorSpec) in the child process.  "ClientRunner" manage the client job execution cycle. The "ClientAppRunner" prepares for the environments and create the ClientRunManager and ClientRunner object. They serve different process purposes.

Reply:
Their names are not clear to what they are doing.
We need to re-think these names and make them clearer.

Some suggestions:

Change "ClientAppRunner" -> "ClientRunner" (Since client run is a larger concept)
Change "ClientRunner" -> "ClientJobRunner"
Change "ClientRunManager" -> "WorkerClientEngine"
Change "ClientEngineExecutorSpec" -> "WorkerClientEngineSpec"